src/util/emphasizeSelection.ts

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash'
+import State from '../@types/State'
+import getThoughtById from '../selectors/getThoughtById'
+
+/** Recursively find for saved node content. */
+const findTargetFocusNode = (
+  root: Element | ChildNode | null,
+  savedFocusNodeContents: string,
+  savedFocusOffset: number,
+): ChildNode | undefined => {
+  if (!root) return undefined
+
+  const childNodes = Array.from(root.childNodes)
+  const hasTextNode = childNodes.some(node => node.nodeName === '#text')
+  if (hasTextNode) {
+    const focusNode = childNodes.find(node => node.textContent === savedFocusNodeContents)
+    return focusNode
+  } else {
+    return findTargetFocusNode(root.firstChild, savedFocusNodeContents, savedFocusOffset)
+  }
+}
+
+/** Emphasize a selection or entire thought.
+ * To be used inside shortcuts of bold/italics/underline.
+ */
+const emphasizeSelection = (state: State, command: 'bold' | 'italic' | 'underline'): void => {
+  const thought = getThoughtById(state, _.last(state.cursor)!)
```

Comment:
You can use `pathToThought` here.

FYI There are a set of standard named selectors that help convert between paths, thoughts, and contexts:

- `thoughtToContext`
- `thoughtToPath`
- `pathToContext`
- `pathToThought`
- `contextToPath`
- `contextToThought`

Avoid the `context*` functions when possible. We generally don't need to use contexts now that each thought has a unique id.